src/DssDirectDepositHub.sol

```diff
@@ -0,0 +1,467 @@
+
+        if (vat.live() == 0) {
+            uint256 culled = ilks[ilk_].culled;
+            // MCD caged
+            require(end.debt() == 0, "DssDirectDepositHub/end-debt-already-set");
+            require(culled == 0, "DssDirectDepositHub/module-has-to-be-unculled-first");
+            _unwind(
+                ilk_,
+                pool,
+                type(uint256).max,
+                availableAssets,
+                Mode.MCD_CAGED
+            );
+        } else if (live == 0) {
+            uint256 culled = ilks[ilk_].culled;
+            // This module caged
+            _unwind(
+                ilk_,
+                pool,
+                type(uint256).max,
+                availableAssets,
+                culled == 1
+                ? Mode.MODULE_CULLED
+                : Mode.NORMAL
+            );
+        } else {
+            // Normal path
+            (uint256 totalAssets, uint256 targetAssets) = pool.calcSupplies(availableAssets);
+
+            if (targetAssets > totalAssets) {
+                _wind(ilk_, pool, targetAssets - totalAssets);
+            } else if (targetAssets < totalAssets) {
+                _unwind(
+                    ilk_,
+                    pool,
+                    totalAssets - targetAssets,
+                    availableAssets,
+                    Mode.NORMAL
+                );
+            }
+        }
+    }
+
+    // --- Collect Interest ---
+    function reap(bytes32 ilk_) external {
+        D3MPoolLike pool = ilks[ilk_].pool;
+
+        require(vat.live() == 1, "DssDirectDepositHub/no-reap-during-shutdown");
+        require(live == 1, "DssDirectDepositHub/no-reap-during-cage");
+
+        uint256 assetBalance = pool.assetBalance();
+        (, uint256 daiDebt) = vat.urns(ilk_, address(pool));
+        if (assetBalance > daiDebt) {
+            uint256 fees = assetBalance - daiDebt;
+            uint256 availableAssets = pool.maxWithdraw();
+            if (fees > availableAssets) {
+                fees = availableAssets;
+            }
+            pool.withdraw(fees);
+            daiJoin.join(address(this), fees);
+            vat.move(address(this), vow, _mul(RAY, fees));
+            emit Reap(ilk_, fees);
+        }
+    }
+
+    // --- Collect any rewards ---
+    function collect(bytes32 ilk_, address[] memory assets, uint256 amount) external returns (uint256 amt) {
+        D3MPoolLike pool = ilks[ilk_].pool;
+
+        amt = pool.collect(assets, amount);
+        emit Collect(ilk_, pool.king(), assets, amt);
+    }
+
+    // --- Allow DAI holders to exit during global settlement ---
+    function exit(bytes32 ilk_, address usr, uint256 wad) external {
+        require(wad <= 2 ** 255, "DssDirectDepositHub/overflow");
+        vat.slip(ilk_, msg.sender, -int256(wad));
+        D3MPoolLike pool = ilks[ilk_].pool;
+        require(pool.transferShares(usr, pool.convertToShares(wad)), "DssDirectDepositHub/failed-transfer");
```

Comment:
Should we use `pool.withdraw()` and avoid the conversion? (this means we add a receiver field to withdraw() but I think it's fine as it appears also in eip-4626).

Reply:
Yeah I like this.

Reply:
what that is doing right now is sending the shares (i.e. adai) to the usr in exchange for the fake collateral we are using. do we want the user to receive Dai instead?

Reply:
right.